octodns/record.py

```diff
@@ -207,9 +207,15 @@ def validate(cls, name, data):
         values = []
         try:
             values = data['values']
+            if values is None or len(values) == 0:
+                values = []
+                reasons.append('missing value(s)')
         except KeyError:
             try:
                 values = [data['value']]
+                if values[0] is None or len(values) == 0:
```

Comment:
This makes sense. Will be adding this.

One question: If we know the value is empty, eg. `''`, why let it through? In the case of an A record, it will just get a `  - invalid ip address ""` reason added. My first reaction is it shouldn't be let through as you've already marked it as an empty value.

Would love to hear another opinion though.

Reply:
> One question: If we know the value is empty, eg. '', why let it through? In the case of an A record, it will just get a - invalid ip address "" reason added.

My thinking was that A/AAAA would be caught by the IPAddress check, but other things that use `_ValueMixin` or `_ValuesMixin`, e.g. `TXT`, wouldn't check for empty. Some of them would get caught by other checks like no trailing `.`. In the end it seemed to be the safest & most consistent to check for empty explicitly. 

> In _ValueMixin, by doing if value it's not letting a None or empty value through, whereas in _ValuesMixin it would be on each individual value

👍 , `if value`'s check for truthy would be fine. A similar check in a foreach on `_ValuesMixing` would work there.